js/src/thankyou.js

```diff
@@ -0,0 +1,354 @@
+            this.modal.modal('show');
+        } else {
+            this.modal.modal('hide');
+        }
+    };
+
+    ThankYou.prototype.showDeleteModal = function (show) {
+        if (show === true) {
+            this.delete_modal.modal('show');
+        } else {
+            this.delete_modal.modal('hide');
+        }
+    };
+
+    ThankYou.prototype.lockThanked = function (lock) {
+        var picker = this.getThankedInput();
+        if (lock) {
+            this.displayPicker(false);
+
+            var thanked_names = '';
+            var options = cla_multi_object_picker.GetSelected(picker);
+            for (var i in options) {
+                thanked_names += options[i].text();
+            }
+            this.setPreselected(thanked_names);
+            this.displayThankedNames(true);
+        } else {
+            this.displayThankedNames(false);
+            this.setPreselected('');
+            this.displayPicker(true);
+        }
+    };
+
+    ThankYou.prototype.displayPicker = function (display) {
+        var picker_container = this.form.find('.js-thank_you-picker-container');
+        if (display === true) {
+            picker_container.show();
+        } else {
+            picker_container.hide();
+        }
+    };
+
+    ThankYou.prototype.displayThankedNames = function (show) {
+        var thanked_names_container = this.form.find('.js-thank_you-thanked-names-container');
+        if (show === true) {
+            thanked_names_container.show();
+        } else {
+            thanked_names_container.hide();
+        }
+    };
+
+    ThankYou.prototype.submit = function () {
+        var self = this;
+
+        self.resetErrors();
+
+        var id = self.getThankYouIDInput().val();
+
+        if (id === '') {
+            id = null;
+        }
+
+        var thanked = self.getThankedInput().val();
+        var tags = self.getTagsInput().val();
+        var description = self.getDescriptionInput().val();
+
+        var body = {
+            description: description
+        };
+
+        if (typeof thanked === 'object') {
+            var thanked_array = [];
+            for (var offset in thanked) {
+                var thanked_split = thanked[offset].split('_');
+                thanked_array.push({oclass: parseInt(thanked_split[0]), id: parseInt(thanked_split[1])});
+            }
+            body.thanked = thanked_array;
+        }
+
+        if (typeof tags === 'object') {
+            body.tags = [];
+
+            if (tags !== null) {
+                for (var offset in tags) {
+                    body.tags.push(parseInt(tags[offset]));
+                }
+            }
+        }
+
+        var url = '/api/thankyou/v2/thankyou';
+        if (id !== null) {
+            url += '/' + id;
+        }
+
+
+        $.ajax({
+            url: url,
+            type: 'POST',
+            dataType: 'json',
+            contentType: 'application/json',
+            data: JSON.stringify(body),
```

Comment:
Doesn't need stringifying, jQuery will do this for you

Reply:
Fixed in 9943b657578350c4ff8824119fbe52fe1eb6c88d

Reply:
Reverted in 76761fba980467b0033f22b6a837819d237739b4 as this caused an invalid JSON response error.

Reply:
Weird, I thought jQuery did this. Guess not! 🤷‍♂